pkg/openstack/loadbalancer.go

```diff
@@ -1137,8 +1128,67 @@ func (lbaas *LbaasV2) ensureOctaviaPool(lbID string, listener *listeners.Listene
 	return pool, nil
 }
 
+func (lbaas *LbaasV2) buildPoolCreateOpt(lbName string, listenerProtocol string, port corev1.ServicePort, service *corev1.Service, svcConf *serviceConfig) v2pools.CreateOpts {
+	// By default, use the protocol of the listener
+	poolProto := v2pools.Protocol(listenerProtocol)
+	if svcConf.enableProxyProtocol {
+		poolProto = v2pools.ProtocolPROXY
+	} else if svcConf.keepClientIP && poolProto != v2pools.ProtocolHTTP {
+		klog.V(4).Infof("Forcing to use %q protocol for pool because annotation %q is set", v2pools.ProtocolHTTP, ServiceAnnotationLoadBalancerXForwardedFor)
+		poolProto = v2pools.ProtocolHTTP
+	}
+
+	affinity := service.Spec.SessionAffinity
+	var persistence *v2pools.SessionPersistence
+	switch affinity {
+	case corev1.ServiceAffinityNone:
+		persistence = nil
+	case corev1.ServiceAffinityClientIP:
+		persistence = &v2pools.SessionPersistence{Type: "SOURCE_IP"}
+	}
+
+	name := lbaas.buildObjectName("pool", lbName, port)
+
+	lbmethod := v2pools.LBMethod(lbaas.opts.LBMethod)
+	return v2pools.CreateOpts{
+		Name:        name,
```

Comment:
I don't think we really need to specify pool name here, like what I mentioned before, the name is gonna be complex. If the user wants to filter pools belongs to a load balancer, just use `--loadbalancer` in CLI.